blocksuite/framework/std/src/inline/range/range-binding.ts

```
@@ -279,6 +279,20 @@ export class RangeBinding {
         selection.is(TextSelection)
       ) ?? null;
 
+    if (!text && selections.length > 0) {
+      const hasRecoverable = selections.find(selection => {
+        const constructor = selection.constructor as typeof BaseSelection;
```

Biome (1.9.4)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)